### ISSUE TITLE
Optional log file

### ISSUE THREAD

**Issue by AlistairNWard** [atks/vt#52]
Would it be possible to (optionally) output all log information to a defined file? We currently catch everything going to stderr and process to identify potential errors in our pipeline. Current implementation means we always have content in stderr and it is almost never an error. What would be great would be a --log <file> option for all vt tools, so that log information goes to that file, errors go to stderr and the vcf/bcf goes to stdout.

**Comment by atks** [comment]
Would it make sense that the --log have an accompanying specified log file name?

Similar to plink's system of specifying the log file thought the --out option.  Except in this case, the default is std::cerr rather than plink.log.

**Comment by AlistairNWard** [comment]
Yes, what I should have written was --log <filename> in the original text. If this isn't supplied, behaviour would be as it is now, but if it is, it keeps that information separate and cleans up the stderr.

**Comment by atks** [comment]
ok.  I'll keep this in mind.  Implementation will come hopefully before the end of the year as I don't have the bandwidth to add this now.

**Comment by AlistairNWard** [comment]
Thanks Adrian, just let me know when you get to it. It's pretty important for us, so we'll want to adopt it as soon as available.

Also, apologies, github was removing some of the text I put in. I meant --log [file], but was using angled brackets, and so the file bit kept getting removed!!

**Comment by atks** [comment]
Can you let me know which tools do you use in vt?  I should be able to implement for those few first.

**Comment by AlistairNWard** [comment]
We use normalize, subset and filter most.

**Comment by atks** [comment]
This will take a while, my code structure is such that it is terrible for systematic logging.  I need to think through the changes.  I'll aim to complete this before the end of the year.  Sorry.

**Comment by AlistairNWard** [comment]
That's ok, Adrian. I didn't post the comment with the expectation that you'd immediately jump on it!
